payments-core/src/main/java/com/stripe/android/model/PaymentMethodCreateParams.kt

```diff
@@ -189,36 +194,43 @@ data class PaymentMethodCreateParams internal constructor(
             }.orEmpty()
         }
 
-    internal enum class Type(
-        private val paymentMethodType: PaymentMethod.Type,
-        val hasMandate: Boolean = false
-    ) {
-        Card(PaymentMethod.Type.Card),
-        Ideal(PaymentMethod.Type.Ideal, hasMandate = true),
-        Fpx(PaymentMethod.Type.Fpx),
-        SepaDebit(PaymentMethod.Type.SepaDebit, hasMandate = true),
-        AuBecsDebit(PaymentMethod.Type.AuBecsDebit, hasMandate = true),
-        BacsDebit(PaymentMethod.Type.BacsDebit, hasMandate = true),
-        Sofort(PaymentMethod.Type.Sofort, hasMandate = true),
-        P24(PaymentMethod.Type.P24),
-        Bancontact(PaymentMethod.Type.Bancontact, hasMandate = true),
-        Giropay(PaymentMethod.Type.Giropay),
-        Eps(PaymentMethod.Type.Eps, hasMandate = true),
-        Oxxo(PaymentMethod.Type.Oxxo),
-        Alipay(PaymentMethod.Type.Alipay),
-        GrabPay(PaymentMethod.Type.GrabPay),
-        PayPal(PaymentMethod.Type.PayPal),
-        AfterpayClearpay(PaymentMethod.Type.AfterpayClearpay),
-        Upi(PaymentMethod.Type.Upi),
-        Netbanking(PaymentMethod.Type.Netbanking),
-        Blik(PaymentMethod.Type.Blik),
-        WeChatPay(PaymentMethod.Type.WeChatPay);
-
-        internal val code = paymentMethodType.code
+    internal sealed class Type : Parcelable {
+        abstract val code: String
```

Comment:
As commented above, can we just introduce a `fromCode(String): Type` instead? The current change also unlinks `PaymentMethod.Type` and `PaymentMethodCreateParams.Type`, which I think should be merged into one later

Reply:
Yes, I agree it's better to keep them linked. We want the mandate data to come from the forms, but I'll make that a separate refactor.